app/src/main/java/com/kickstarter/viewmodels/ChangePasswordViewModel.kt

```diff
@@ -0,0 +1,176 @@
+                    .compose(takeWhen<ChangePassword, Void>(this.changePasswordClicked))
+                    .switchMap { cp -> submit(cp).materialize() }
+                    .compose(bindToLifecycle())
+                    .share()
+
+            changePasswordNotification
+                    .compose(errors())
+                    .subscribe({ this.error.onNext(it.localizedMessage) })
+
+            changePasswordNotification
+                    .compose(values())
+                    .map { it.updateUserAccount()?.user()?.email() }
+                    .subscribe(this.success)
+
+        }
+
+        private fun submit(changePassword: ChangePasswordViewModel.ViewModel.ChangePassword): Observable<UpdateUserPasswordMutation.Data> {
+            return this.apolloClient.updateUserPassword(changePassword.currentPassword, changePassword.newPassword, changePassword.confirmPassword)
+                    .doOnSubscribe { this.progressBarIsVisible.onNext(true) }
+                    .doAfterTerminate { this.progressBarIsVisible.onNext(false) }
+        }
+
+        override fun changePasswordClicked() {
+            this.changePasswordClicked.onNext(null)
+        }
+
+        override fun confirmPassword(confirmPassword: String) {
+            this.confirmPassword.onNext(confirmPassword)
+        }
+
+        override fun currentPassword(currentPassword: String) {
+            this.currentPassword.onNext(currentPassword)
+        }
+
+        override fun newPassword(newPassword: String) {
+            this.newPassword.onNext(newPassword)
+        }
+
+        override fun error(): Observable<String> {
+            return this.error
+        }
+
+        override fun passwordConfirmationWarningIsVisible(): Observable<Boolean> {
+            return this.passwordConfirmationWarningIsVisible
+        }
+
+        override fun passwordLengthWarningIsVisible(): Observable<Boolean> {
+            return this.passwordLengthWarningIsVisible
+        }
+
+        override fun progressBarIsVisible(): Observable<Boolean> {
+            return this.progressBarIsVisible
+        }
+
+        override fun saveButtonIsEnabled(): Observable<Boolean> {
+            return this.saveButtonIsEnabled
+        }
+
+        override fun success(): Observable<String> {
+            return this.success
+        }
+
+        data class ChangePassword(val currentPassword: String, val newPassword: String, val confirmPassword: String) {
+            fun isValid(): Boolean {
+                return isNotEmptyAndAtLeast6Chars(this.currentPassword)
+                        && isNotEmptyAndAtLeast6Chars(this.newPassword)
+                        && isNotEmptyAndAtLeast6Chars(this.confirmPassword)
+                        && this.confirmPassword == this.newPassword
+            }
+
+            private fun isNotEmptyAndAtLeast6Chars(password: String) = !password.isEmpty() && password.length >= MINIMUM_PASSWORD_LENGTH
+        }
+    }
+
+    companion object {
```

Comment:
You can define this at the top of the class I think instead of a companion object because no other class is accessing this constant

```
const val MINIMUM_PASSWORD_LENGTH = 6

interface ChangePasswordViewModel {

```